pkg/admissioncontroller/webhooks/admission/auditpolicy/admission.go

```diff
@@ -0,0 +1,285 @@
+	}
+
+	// Validate audit policy semantically
+	auditPolicyObj, schemaVersion, err := decoder.Decode([]byte(auditPolicy), nil, nil)
+	if err != nil {
+		return admission.Errored(http.StatusUnprocessableEntity, fmt.Errorf("failed to decode the provided audit policy err=%v", err))
+	}
+	auditPolicyInternal, ok := auditPolicyObj.(*audit_internal.Policy)
+	if !ok {
+		return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failure to cast to audit Policy type: %v", schemaVersion))
+	}
+	errList := auditvalidation.ValidatePolicy(auditPolicyInternal)
+	if len(errList) != 0 {
+		return admission.Errored(http.StatusUnprocessableEntity, fmt.Errorf("provided invalid audit policy err=%v", errList))
+	}
+
+	// Validate schema version against Shoot Kubernetes version
+	shootList := &gardencorev1beta1.ShootList{}
+	if err := h.apiReader.List(ctx, shootList, client.InNamespace(request.Namespace)); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	if err := validatePolicyAgainstShoots(request, schemaVersion, shootList); err != nil {
+		return admission.Errored(http.StatusUnprocessableEntity, err)
+	}
+
+	return acadmission.Allowed("configmap change is valid")
+}
+
+// IsValidAuditPolicyVersion checks whether the api server support the provided audit policy apiVersion
+func IsValidAuditPolicyVersion(shootVersion string, schemaVersion *schema.GroupVersionKind) (bool, error) {
+	auditGroupVersion := schemaVersion.GroupVersion().String()
+
+	//TODO Update check once https://github.com/kubernetes/kubernetes/issues/98035 is resolved
+	if auditGroupVersion == "audit.k8s.io/v1" {
+		return version.CheckVersionMeetsConstraint(shootVersion, ">= v1.12")
+	}
+	return true, nil
+}
+
+func (h *handler) getOldObject(request admission.Request, oldObj runtime.Object) error {
+	if len(request.OldObject.Raw) != 0 {
+		return h.decoder.DecodeRaw(request.OldObject, oldObj)
+	}
+	return fmt.Errorf("could not find old object")
+}
+
+func getAuditPolicy(cm *corev1.ConfigMap) (auditPolicy string, err error) {
+	auditPolicy, ok := cm.Data[auditPolicyConfigMapDataKey]
+	if !ok {
+		return "", fmt.Errorf("missing '.data.policy' in audit policy configmap")
+	}
+	if len(auditPolicy) == 0 {
+		return "", fmt.Errorf("empty audit policy. Provide non-empty audit policy")
+	}
+	return auditPolicy, nil
+}
+
+func validatePolicyAgainstShoots(request admission.Request, schemaVersion *schema.GroupVersionKind, shoots *gardencorev1beta1.ShootList) error {
+	for _, s := range shoots.Items {
+		if hasAuditPolicy(&s) {
+			if s.Spec.Kubernetes.KubeAPIServer.AuditConfig.AuditPolicy.ConfigMapRef.Name == request.Name {
+				isValidVersion, err := IsValidAuditPolicyVersion(s.Spec.Kubernetes.Version, schemaVersion)
+				if err != nil {
+					return err
+				}
+				if !isValidVersion {
+					return fmt.Errorf("your shoot cluster version %q is not compatible with audit policy version %q", s.Spec.Kubernetes.Version, schemaVersion.GroupVersion().String())
```

Comment:
It would be helpful to include the shoot name with the incompatible K8s version in the error message.